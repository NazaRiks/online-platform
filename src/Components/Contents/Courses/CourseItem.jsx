export  default function CourseIem(){
    return (<div></div>)
}