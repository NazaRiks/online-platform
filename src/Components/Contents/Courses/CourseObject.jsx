import React, { useState } from "react";
import "./Styles/CourseObject.css";
import courseData from "../../../Data's/courseData.js";
import { useLocation } from 'react-router-dom';
export default function CourseObject() {
    const location = useLocation();
    const { id, countOfLessons, author, price, userCourseCount, description, mainDescription, title, image, lessons } = location.state;
    const paid = courseData[id].paid;
    const [alreadyPaid, setAlreadyPaid] = useState(paid ? "Ви вже придаби даний курс" : "Придбати");
    const AddPayment = (id) => {

        const index = courseData.findIndex((item) => item.id === id);
        if (index !== -1) {
            courseData[index].paid = true;
            console.log(courseData[index].sub);
        } else {
            console.error("Елемент з id ${id} не знайдено в масиві courseData");
        }
    };

    const handleActionButtonClick = () => {
        AddPayment(id);
        setAlreadyPaid("Ви вже придбали даний курс");
    };
    const lessonsContent = paid ? <div className="course-object-lessons">
        <h2>Почни навчання прямо зараз! </h2>
        {lessons.map((item) => (
            <div key={item.id}>
                <button>{item.title}</button>
            </div>
        ))}
    </div> : null

    return (
        <div className="course-object">
            <div className="course-object-left-details">
                <h2>{title}</h2>
                <p>Автор:{author}</p>
                <p>Кількість уроків{countOfLessons}</p>
                <p>{mainDescription}</p>
            </div>
            <div className="course-object-right-details">
                <img src={image} alt={title}/>
                <p>{description}</p>
                <p>Кількість учасників:{userCourseCount}</p>
                <p>Ціна: {price}</p>
                <div className="course-object-acion">
                    <button disabled={paid} onClick={handleActionButtonClick}>{alreadyPaid}</button>
                </div>
            </div>
            {lessonsContent}
        </div>
    );
}