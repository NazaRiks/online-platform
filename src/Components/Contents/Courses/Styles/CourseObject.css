.course-object {
    display: flex;
    flex-wrap: wrap;
    margin-bottom: 20px;
    border: 1px solid #ddd;
    border-radius: 5px;
}
.actual-course{
    text-align: center;
}
.course-object-details {
    display: flex;
    width: 100%;
}

.course-object-left-details {
    flex: 1;
    padding: 20px;
}

.course-object-right-details {
    flex: 1;
    padding: 20px;
    text-align: center;
}

.course-object-right-details img {
    width: 100%;
}

.course-object-action {
    margin-top: 10px;
}
.course-object-acion button{
    padding: 10px 15px;
    border: 1px solid #ddd;
    border-radius: 5px;
    background-color: #45a049;
    cursor: pointer;
    transition: all 0.2s ease-in-out;
    margin: 5px;
    font-size: 1.2rem;
    color:white;

}


.course-object-lessons {
    width: 100%;
    padding: 20px;
}

.course-object-lessons button {
    padding: 10px 15px;
    border: 1px solid #ddd;
    border-radius: 5px;
    background-color: #45a049;
    cursor: pointer;
    transition: all 0.2s ease-in-out;
    margin: 5px;
    font-size: 1.2rem;
    color:white;
    width: 100%;
}


.course-object-lessons button:hover {
    background-color: #357834;
}
