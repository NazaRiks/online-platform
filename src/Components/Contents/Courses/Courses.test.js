import { describe, test, expect, mount } from 'vitest';
import MyCoursePage from "./CoursePage.jsx";
import CourseObject from "./CourseObject.jsx";
import CourseListItems from "./CourseListItems.jsx";
import CourseItem from "./CourseItem.jsx";

describe('courses test page', () => {
    test('course page', () => {
    const data = { foo: new Set(['bar', 'snapshot']) }
    expect(data).toMatchSnapshot({ foo: expect.any(Set) })
})
    test('couse object', () => {
        const data = { foo: new Set(['bar', 'snapshot']) }
        expect(data).toMatchSnapshot({ foo: expect.any(Set) })
    })
    test('course list items', () => {
        const data = { foo: new Set(['bar', 'snapshot']) }
        expect(data).toMatchSnapshot({ foo: expect.any(Set) })
    })
    test('course-item', () => {
        const data = { foo: new Set(['bar', 'snapshot']) }
        expect(data).toMatchSnapshot({ foo: expect.any(Set) })
    })
})
test('handles course selection correctly', () => {
    const div = document.createElement('div');
    const mockCourseData = [
        {
            id: 1,
            image: 'sample.jpg',
            description: 'Sample description',
            title: 'Sample Course 1',
            author: 'John Doe',
            price: 49.99,
            userCourseCount: 100,
            mainDescription: 'Sample main description',
            lessons: [{ id: 1, title: 'Lesson 1' }, { id: 2, title: 'Lesson 2' }],
            countOfLessons: 2
        },
        // Add more sample data as needed
    ];

    const myCoursePageInstance = mount(MyCoursePage, div);

    // Mock the useNavigate hook
    // eslint-disable-next-line no-undef
    jest.spyOn(require('react-router-dom'), 'useNavigate').mockReturnValueOnce(jest.fn());

    // Simulate a course selection
    myCoursePageInstance.handleChoose(mockCourseData[0].id);

    // Access the internal state or verify side effects as needed
    const navigateMock = require('react-router-dom').useNavigate();
    // For example: expect(navigateMock).toHaveBeenCalledWith('/courses/course-object', { state: queryParams });
});
describe('CourseObject Component', () => {
    test('renders without crashing', () => {
        const div = document.createElement('div');
        mount(CourseObject, div, {
            location: {
                state: {
                    id: 1,
                    countOfLessons: 10,
                    author: 'John Doe',
                    price: 49.99,
                    userCourseCount: 100,
                    description: 'Sample description',
                    mainDescription: 'Sample main description',
                    title: 'Sample Course',
                    image: 'sample.jpg',
                    lessons: [{ id: 1, title: 'Lesson 1' }, { id: 2, title: 'Lesson 2' }],
                },
            },
        });
        expect(div.innerHTML).toMatchSnapshot();
    });
})
describe('CourseListItems Component', () => {
    const mockCourseData = [
        {
            id: 1,
            countOfLessons: 10,
            author: 'John Doe',
            price: 49.99,
            userCourseCount: 100,
            description: 'Sample description',
            title: 'Sample Course 1',
            image: 'sample1.jpg',
            paid: false,
        },
        {
            id: 2,
            countOfLessons: 8,
            author: 'Jane Smith',
            price: 29.99,
            userCourseCount: 80,
            description: 'Another description',
            title: 'Sample Course 2',
            image: 'sample2.jpg',
            paid: true,
        },
        // Add more sample data as needed
    ];

    test('renders without crashing', () => {
        const div = document.createElement('div');
        // eslint-disable-next-line no-undef
        mount(CourseListItems, div, { courseData: mockCourseData, onAction: jest.fn() });
        expect(div.innerHTML).toMatchSnapshot();
    });
    })
test('calls onAction correctly', () => {
    const div = document.createElement('div');
    // eslint-disable-next-line no-undef
    const mockOnAction = jest.fn();
    const mockCourseItem = {
        id: 1,
        countOfLessons: 10,
        author: 'John Doe',
        price: 49.99,
        userCourseCount: 100,
        description: 'Sample description',
        title: 'Sample Course 1',
        image: 'sample1.jpg',
        paid: false,
    };
    const courseItemInstance = mount(CourseItem, div, { ...mockCourseItem, onAction: mockOnAction });

    // Simulate interaction with the button
    courseItemInstance.querySelector('button').click();

    // Verify if onAction function was called with the correct arguments
    expect(mockOnAction).toHaveBeenCalledWith(mockCourseItem.id);
});