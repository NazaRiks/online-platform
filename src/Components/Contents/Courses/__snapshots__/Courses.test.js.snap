// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`courses test page > course list items 1`] = `
{
  "foo": Any<Set>,
}
`;

exports[`courses test page > course page 1`] = `
{
  "foo": Any<Set>,
}
`;

exports[`courses test page > course-item 1`] = `
{
  "foo": Any<Set>,
}
`;

exports[`courses test page > couse object 1`] = `
{
  "foo": Any<Set>,
}
`;

exports[`forum page > forum list items 1`] = `
{
  "foo": Any<Set>,
}
`;

exports[`forum page > forum object 1`] = `
{
  "foo": Any<Set>,
}
`;

exports[`forum page > forum page 1`] = `
{
  "foo": Any<Set>,
}
`;

exports[`forum page > forum-item 1`] = `
{
  "foo": Any<Set>,
}
`;
